webforces/server/structs.py

```diff
@@ -36,13 +36,28 @@ class Algorithm:
     tests_id: List[int] = field(default_factory=list)
     lang_id: int = 0
 
+    @classmethod
+    def fromDict(cls, dict):
+        return cls(
+            dict["alg_id"], dict["title"],
+            dict["author_id"], dict["source"], dict["tests_id"],
+            dict["lang_id"]
+        )
+
 
 @dataclass
 class Test:
     test_id: int
+    alg_title: str = ''
     title: str = ''
```

Comment:
Why do we have both `title` and `alg_title`? What is the difference? If we need both, please, add comments that describe the difference.

Reply:
title == test_title.

Reply:
Having title for algorithm doesn't make any sense. I agree that we need to connect test and algorithm. Maybe it's better to connect them using alg_id?

Reply:
Alg_id is not unique. It is unique within the user.

There is two ways to connect test and alg:
1. Add alg_title to test structure (alg_title is unique);
2. Add author_id and alg_id to test structure (author_id is unique, but it does not connect test and alg by itself).

So, I believe that the first way is better.